h2o-algos/src/main/java/hex/rulefit/RuleFitModel.java

```diff
@@ -173,4 +177,39 @@ public RuleFitMojoWriter getMojo() {
     public boolean haveMojo() {
         return true;
     }
+    
+    @Override
+    public Frame fitRules(Frame frame, String[] ruleIds) {
+        Frame adaptFrm = new Frame(frame);
+        adaptTestForTrain(adaptFrm, true, false);
+        
+        List<Rule> rules = new ArrayList<>();
+        List<String> linearRules = new ArrayList<>();
+        for (int i = 0; i < ruleIds.length; i++) {
+            if (ruleIds[i].startsWith("linear.") && isLinearVar(ruleIds[i])) {
+                linearRules.add(ruleIds[i]);
+            } else {
+            rules.add(ruleEnsemble.getRuleByVarName(RuleFitUtils.readRuleId(ruleIds[i])));
+            }
+        }
+        RuleEnsemble subEnsemble = new RuleEnsemble(rules.toArray(new Rule[0]));
+        Frame result = subEnsemble.transform(adaptFrm);
+        // linear rules apply to all the rows
+        for (int i = 0; i < linearRules.size(); i++) {
+            result.add(linearRules.get(i), Vec.makeOne(frame.numRows()));
+        }
+        
+        result = new Frame(Key.make(), result.names(), result.vecs());
+        DKV.put(result);
+        return result;
+    }
+    
+    private boolean isLinearVar(String potentialLinVarId) {
+        List<String> linVarNames = Arrays.asList(glmModel.names()).stream().filter(name -> name.startsWith("linear.")).collect(Collectors.toList());
```

Comment:
IMO the list `linVarNames` could be initialized only once, or am I missing something?

Reply:
thats a good point, thanks!

Reply:
8902392